web/src/engine/websites/DisasterScans.ts

```diff
@@ -51,34 +41,32 @@ export default class extends DecoratableMangaScraper {
         return icon;
     }
 
-    public override async Initialize(): Promise<void> {
-        const data = await FetchWindowScript<NEXTDATA>(new Request(this.URI), `__NEXT_DATA__`);
-        this.nextBuild = data.buildId;
-    }
-
-    public override ValidateMangaURL(url: string): boolean {
-        return new RegExpSafe(`^${this.URI.origin}/comics/[^/]+`).test(url);
-    }
-
-    public override async FetchManga(provider: MangaPlugin, _url: string): Promise<Manga> {
-        const slug = _url.split('/').at(-1);
-        const url = new URL(`/_next/data/${this.nextBuild}/comics/${slug}.json?slug=${slug}`, this.URI);
-        const data = await FetchJSON<JSONManga>(new Request(url));
-        return new Manga(this, provider, data.pageProps.comic.id, data.pageProps.comic.ComicTitle.trim());
+    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
+        const scripts = await FetchCSS<HTMLScriptElement>(new Request(new URL(`${manga.Identifier}`, this.URI)), 'script:not([src])');
+        const chapters = this.ExtractData<JSONChapter[]>(scripts, 'ChapterNumber', 'chapters');
+        return chapters.map(chapter => {
+            let title = `Chapter ${chapter.ChapterNumber}`;
+            title += chapter.ChapterName ? ` - ${chapter.ChapterName}` : '';
+            return new Chapter(this, manga, `${manga.Identifier}/${chapter.chapterID}-chapter-${chapter.ChapterNumber}`, title);
+        });
     }
 
-    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
-        const data = await FetchWindowScript<JSONMangas>(new Request(new URL('/comics', this.URI)), `__NEXT_DATA__.props`);
-        return data.pageProps.comics.map(element => new Manga(this, provider, element.id, element.ComicTitle.trim()));
-    }
+    private ExtractData<T>(scripts: HTMLScriptElement[], scriptMatcher: string, keyName: string): T {
+        const script = scripts.map(script => script.text).find(text => text.includes(scriptMatcher) && text.includes(keyName));
+        const content = JSON.parse(script.substring(script.indexOf(',"') + 1, script.length - 2)) as string;
+        let record = JSON.parse(content.substring(content.indexOf(':') + 1)) as JSONObject;
 
-    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
-        const slug = `${manga.Identifier}-${manga.Title.toLowerCase().split(' ').join('-')}`;
-        const url = new URL(`/_next/data/${this.nextBuild}/comics/${slug}.json?slug=${slug}`, this.URI);
-        const data = await FetchJSON<JSONManga>(new Request(url));
-        return data.pageProps.chapters.map(chap => {
-            const title = `Chapter ${chap.chapterNumber}${chap.ChapterName.length != 0 ? ' - ' + chap.ChapterName : ''}`;
-            return new Chapter(this, manga, `/comics/${slug}/${chap.chapterID}-chapter-${chap.chapterNumber}`, title);
-        });
+        return (function FindValueForKeyName(parent: JSONElement): JSONElement {
```

Comment:
If it is intended to use more JSON queries in the future it might be worth evaluating libraries such as [jsonpath-plus](https://www.npmjs.com/package/jsonpath-plus) instead of re-implementing multiple a custom functions.